### PR TITLE
Create notification when moderator access of user is changed in a group

### LINKED ISSUE #59: Notifications
**Note: Once, tables and endpoints for both notifications and groups have been created, make sure that the following notifications are generated automatically:**

- [ ] When users are added to a non-confidential group:
```
{
    heading : 'Added to group ${group-name}',
    description: `You have been added to the group [${group-name}](http://${FRONTEND_HOST}:${FRONTEND_PORT}/groups/${group-id})` ,
}
```
- [x] When a user is added as a moderator:
```
{
    heading : 'Promoted to moderator of ${group-name}',
    description: `You have been promoted to moderator of the group [${group-name}](http://${FRONTEND_HOST}:${FRONTEND_PORT}/groups/${group-id})` ,
}
```
- [x] When a user is removed from a moderator:
```
{
    heading : 'Removed from moderator of ${group-name}',
    description: `You are no longer a moderator of the group [${group-name}](http://${FRONTEND_HOST}:${FRONTEND_PORT}/groups/${group-id})` ,
}
```

### PR DESCRIPTION
**Pull request by cjchirag7**
Fixes #59 

- I have skipped the notifications when users are added to a non-confidential group, as it seems to run a large number of queries. We can think for that later.

### CODE REVIEW

File: server/models/groups/removeModerator.js [hunk 1]
```diff
@@ -52,7 +52,21 @@ function removeModerator({ params, body, username }) {
                     'Either the person is already not a moderator or no such person found in the group!'
                   )
                 }
-                return resolve('Successfully removed from moderator!!')
+                const port = process.env.FRONTEND_PORT || 3000
+                const host = process.env.FRONTEND_HOST || 'localhost'
+                const body = {
+                  heading: `Removed from moderator of ${groupName}`,
+                  description: `You are no longer a moderator of the group [${groupName}](http://${host}:${port}/groups/${groupId})`,
```
Comment (ridhishjain) [inline server/models/groups/removeModerator.js:59]: In this, I think the moderator will be able to remove himself (as moderator) ....would that be right?
and one more thing...do we need to show "removed moderator by whom"? (like removed admin by xyz in whatsapp)

**Comment by ridhishjain** [comment]
rest LGTM!!, can merge

File: server/models/groups/removeModerator.js (hunk 1 above)
Comment (cjchirag7) [reply server/models/groups/removeModerator.js:59]: Whatsapp generally shows "You are no longer an admin"

Comment (cjchirag7) [reply server/models/groups/removeModerator.js:59]: For now, we can allow moderator to remove himself (as moderator) . Later we can change it, if required.

**Pull request merged**